notify-api/src/notify_api/core/authentication.py

```diff
@@ -0,0 +1,149 @@
+""" Waiting for starlette release this change
+    https://github.com/encode/starlette/issues/794
+    https://github.com/encode/starlette/pull/942
+ """
+# pylint: skip-file
+# flake8: noqa
+import asyncio
+import functools
+import inspect
+import typing
+
+from starlette.exceptions import HTTPException
+from starlette.requests import HTTPConnection, Request
+from starlette.responses import RedirectResponse, Response
+from starlette.websockets import WebSocket
+
+
+def has_required_scope(conn: HTTPConnection, scopes: typing.Sequence[str]) -> bool:
+    for scope in scopes:
+        if scope not in conn.auth.scopes:
+            return False
+    return True
+
+
+def requires(
+        scopes: typing.Union[str, typing.Sequence[str]],
+        status_code: int = 403,
+        redirect: str = None,
+) -> typing.Callable:
+    scopes_list = [scopes] if isinstance(scopes, str) else list(scopes)
+
+    def decorator(func: typing.Callable) -> typing.Callable:
+        type = None
+        sig = inspect.signature(func)
+        for idx, parameter in enumerate(sig.parameters.values()):
+            if parameter.name == "request" or parameter.name == "websocket":
+                type = parameter.name
+                break
+        else:
+            raise Exception(
+                f'No "request" or "websocket" argument on function "{func}"'
+            )
+
+        if type == "websocket":
```

Comment:
Do we have any web socket integration? Not sure why we need this check here. Can you please add some comments ?

Reply:
This is not our code. I am waiting for new release of starlette package to fix a bug. Once they released we can remove this file.